plugins/flutter_aepmessaging/README.md

```diff
@@ -0,0 +1,271 @@
+
+Refer to the [Initialization](https://github.com/adobe/aepsdk_flutter#initializing) section of the root README for more information about initializing the SDK.
+
+**Initialization Example**
+
+iOS
+
+```objc
+// AppDelegate.h
+@import AEPCore;
+@import AEPEdge;
+@import AEPMessaging;
+...
+@implementation AppDelegate
+
+// AppDelegate.m
+- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+    [AEPMobileCore setWrapperType:AEPWrapperTypeFlutter];
+
+     // TODO: Set up the preferred Environment File ID from your mobile property configured in Data Collection UI
+    NSString* ENVIRONMENT_FILE_ID = @'YOUR-APP-ID';
+
+    NSArray *extensionsToRegister = @[AEPMessaging.class,
+                                      AEPMobileEdge.class
+                                      ];
+
+    [AEPMobileCore registerExtensions:extensionsToRegister completion:^{
+    [AEPMobileCore configureWithAppId: ENVIRONMENT_FILE_ID];
+    }];
+    return YES;
+ }
+```
+
+Android
+
+```java
+import com.adobe.marketing.mobile.MobileCore;
+import com.adobe.marketing.mobile.Edge;
+import com.adobe.marketing.mobile.messaging.Messaging;
+...
+import io.flutter.app.FlutterApplication;
+...
+public class MainApplication extends FlutterApplication {
+  ...
+  // TODO: Set up the preferred Environment File ID from your mobile property configured in Data Collection UI
+   private final String ENVIRONMENT_FILE_ID = "YOUR-APP-ID";
+
+    @Override
+    public void onCreate() {
+      super.onCreate();
+      List<Class<? extends Extension>> extensions = Arrays.asList(   
+              Edge.EXTENSION,
+              Messaging.EXTENSION
+      );
+      MobileCore.registerExtensions(extensions, o -> MobileCore.configureWithAppID(ENVIRONMENT_FILE_ID));
+    }
+}
+```
+
+---
+
+### Importing the extension
+
+In your Flutter application, import the Messaging extension as follows:
+
+```dart
+import 'package:flutter_aepmessaging/flutter_aepmessaging.dart';
+```
+
+---
+
+## API reference
+
+### extensionVersion
+
+Returns the SDK version of the Messaging extension.
+
+**Syntax**
+
+```dart
+static Future<String> get extensionVersion
+```
+
+**Example**
+
+```dart
+String version = await Messaging.extensionVersion;
+```
+
+---
+
+### getCachedMessages
+
+Returns a list of messages that have currently been cached in-memory using `Messaging.saveMessage()`
+
+**Syntax**
+
+```dart
+List<Message> messages = await Messaging.getCachedMessages();
+```
+
+---
+
+### refreshInAppMessages
+
+This API retrieves the Experience Cloud ID (ECID) that was generated when the app was initially launched. This ID is preserved between app upgrades, is saved and restored during the standard application backup process, and is removed at uninstall.
+
+**Syntax**
+
+```dart
+static Future<void> refreshInAppMessages
+```
+
+**Example**
+
+```dart
+await Messaging.refreshInAppMessages();
+```
+
+## Handling In App Messages using Message Object
```

Comment:
We need to make a note that the user need to call getcachedMessages for the message before using these functions.